### ISSUE TITLE
Amazon MSK GA features

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#8823]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Additional features for [Amazon MSK GA](https://aws.amazon.com/about-aws/whats-new/2019/05/amazon_managed_streaming_for_apache_kafka_amazon_msk_is_now_generally_available/).

### [`aws_msk_cluster`](https://www.terraform.io/docs/providers/aws/r/msk_cluster.html)

* Client authentication
* Encryption-in-transit
* [Configuration](https://www.terraform.io/docs/providers/aws/r/msk_configuration.html)

[Broker storage](https://docs.aws.amazon.com/msk/1.0/apireference/clusters-clusterarn-nodes-storage.html) can now be updated but not sure how this fits with the current resource model.

Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/6653
* https://github.com/terraform-providers/terraform-provider-aws/issues/8736

**Comment by bflad** [comment]
Work in progress pull request submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/8850

**Comment by bflad** [comment]
The following has been merged:

```
* data-source/aws_msk_cluster: Add `bootstrap_brokers_tls` attribute
* resource/aws_msk_cluster: Add `bootstrap_brokers_tls` attribute
* resource/aws_msk_cluster: Support `broker_node_group_into` configuration block `ebs_volume_size` argument updates
* resource/aws_msk_cluster: Add `client_authentication` configuration block
* resource/aws_msk_cluster: Add `configuration_info` configuration block
* resource/aws_msk_cluster: Add `current_version` attribute
* resource/aws_msk_cluster: Add `encryption_info` configuration block `encryption_in_transit` configuration block
* resource/aws_msk_cluster: Support tagging on creation
```

And will release with version 2.14.0 of the Terraform AWS Provider, likely tomorrow. 👍

**Comment by bflad** [comment]
This has been released in [version 2.14.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2140-june-06-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
